### PR TITLE
Support custom external plugins

### PR DESCRIPTION
**Pull request by Kirill888**
Basically

```yaml
plugin: somelib.MyPlugin
```

where `somelib.MyPlugin` must derive from `odc.stats.model.StatsPluginInterface`

### CODE REVIEW

**Comment by Kirill888** [comment]
@fivejjs let's switch your plugins to use this method rather than keeping code in this repo.

**Review by alexgleith — APPROVED** [review]
Looks good to me! This is how Alchemist does it, actually all of them are a path to an Object... could do that instead of having internal plugins a special case.

I like it how it is now, anyhow.

File: libs/stats/odc/stats/_plugins.py [hunk 1]
```diff
@@ -14,6 +15,14 @@ def _new(plugin_class, *args, **kwargs) -> StatsPluginInterface:
 
 def resolve(name: str) -> PluginFactory:
     maker = _plugins.get(name)
+    if maker is None:
+        maker = pydoc.locate(name)
+        if maker is not None:
+            mro = getattr(maker, "__mro__", tuple())
```
Comment (omad) [inline libs/stats/odc/stats/_plugins.py]: What's wrong with `issubclass(maker, StatsPluginInterface)`?

**Review by omad — APPROVED** [review]

File: libs/stats/odc/stats/_plugins.py (hunk 1 above)
Comment (Kirill888) [reply libs/stats/odc/stats/_plugins.py]: nothing, could not google it, is all

**Pull request merged**